### ISSUE TITLE
[RFC] Moving forward and priorities

### ISSUE THREAD

**Issue by SleeplessByte** [react-static/react-static#1262]
🚀 Maintenance update releasing after this weekend (week of August 12th). After that we'll work on closing/merging the remaining PRs, and then regroup and figure out what's next from the list below.

----

Hello everyone!

If you have followed #1259, you're aware that [nozzle.io](https://nozzle.io/) no longer actively backs this project. One of the main clients and projects of [XP Bytes](https://xpbytes.com) heavily relies on a company-fork of `react-static`; because I and by proxy the company really would like React Static to be completely Open Source and a community effort, you can see the repository is now owned by the react-static organisation.

For the foreseeable future, I (and perhaps others in my company) will actively contribute and dogfood `react-static` and we'll extract features out of our platform and merge the back into this project, similar to how [Basecamp](https://basecamp.com/) feeds [Ruby on Rails](https://rubyonrails.org/). It will take a while for me, and the rest of the team, to get up to speed with everything currently outstanding, but I am planning to set-up a project board to transparently communicate with you all what we think the next steps forward are.

Right now, I'm listing a few of the current open RFCs (#1199 #1180), that are still considered, will list out a few of my immediate priorities and open up the conversation. I would ❤️ it if everyone who relies on this project could tell me at most three things:

- What current 🐛 **bug** is actively hindering you (issue linking please!).
- What current 🦄 **feature** is currently missing and you really want (issue linking please!).
- What 💎 "nice to have" you think is on the top of the list of things that are "nice to have".

(It's helpful if you use these emojis to indicate what you are referring to).

On my priorities list:

- [x] Merge / Review open PRS 
  - [x] specifically #1219 and fix CI, 
  - [x] #1196 #1250 #1253 #1254 for path handling, 
  - [x] #1222
  - [x] #1226 which fixes #1220 and #1239, and 
  - [x] #1181.
- [x] Review the current codebase
  - [x] consider dropping unused packages
  - [x] consider partially upgrading some to TypeScript (won't affect you at all)
- [x] Get e2e tests up and running, even if it's just a smoke test.
  - [x] fix CI
  - [x] have at least a smoke test for the main package
  - [x] have proper tests for the plugins (!!)
- [x] Consider the next major update to use `@react-static/*` to clean up `node_modules`.
- [ ] Get the `CONTRIBUTING` up to speed so it's _easier_ to contribute.

Let me know what you think! 🚀 🚀 🚀 

**EDIT**: Once this list is complete enough, I'll do a release. I'm discussing with others what the best approach is, but basically I think we'll have one of these RFCs for each major version going forward, so you can all tell me what ya want ;)

Yes, build stabilisation should be worked on as well.

**Comment by david476** [comment]
Hello! No opinions to express, just glad the platform I just decided to build my site on in going to continue being developed. Hopefully I can take on some issues in the coming weeks :)

**Comment by sw-yx** [comment]
thats a good todolist! looking forward to see what you do with it!

**Comment by philhawksworth** [comment]
This is exciting. @tannerlinsley and the Nozzle team have done incredible work on this project. It's nice though to see this increased input from a wider community and some fresh stewardship to give Tanner a chance to take a well-earned breath!

This task list looks great. I'll be sure to voice some input along the way. Thank you for this!

**Comment by digitalkaoz** [comment]
I think this bits are missing:

/ deliver es5/es6 bundles depending on the browser
/ react-static itself shouldnt be compiled down to es5 (es6 is perfectly fine, for the rest we have the final project babelification). That should give us leaner bundles!
/ fix resolving of Lazy Components during SSR (#1238)
/ Unit Tests
/ Integration Tests
/ port Codebase to Typescript

**Comment by CalMlynarczyk** [comment]
@digitalkaoz I like this list. The limited bundling customization was a significant pain point for my org when we built our site.

We also ran into a few stability issues with v6, and upgrading to v7 only exacerbated the problems we had (random build failures, hot reloading not working, incompatibility with packages and our Babel configuration during compilation). We like react-static and have worked through most of these issues, but it was harder than it should have been due to a lot of the documentation not getting updated for v7 (ex. https://github.com/react-static/react-static/blob/master/docs/concepts.md). I will try to get a PR in sometime in the next couple of weeks to address the incorrect docs we found.

@SleeplessByte Is the new Travis setup going to live at https://travis-ci.com/react-static/react-static? I saw you had some commits building there. The [old Travis page](https://travis-ci.org/nozzle/react-static/builds) isn't showing the latest merges to `master`; I assume because of the project owner change.

**Comment by SleeplessByte** [comment]
@CalMlynarczyk yes, Travis is running again. There are quite a few tests that fail (see #1265, https://github.com/react-static/react-static/pull/1250#issuecomment-512852964) because travis hasn't been running for over a month or more.

This #1219 PR fixed that.

**Comment by DDunc** [comment]
I'm grateful for all the work @tannerlinsley has put into this project and I'm also really glad to see the handoff is starting strong! I expect there will be some adjustment period at first but I think some consolidation is going to be a good thing for preserving and growing the existing community.  

To +1 some things that have already been mentioned and articulate a bit more:

* **Documentation and Examples:** The v5 docs were extensive, the docs site worked well and there were a huge number of helpful examples and starter repos within `react-static create` that were fairly stable. Visibly getting back to that level for v7 is important for bringing in new users and making the "upgrade or migrate to a new framework" decision easier to make. Putting http://react-static.js.org or similar back up would be helpful as well.
* **Test Coverage:** More unit and integration tests for existing code will have a lot of benefits for both contributors and users, and could be a great starting point for community members who haven't contributed before and would like to gain a better understanding of the underlying code. This can feed right into improving the documentation for contributing as well. 
* **Comprehensive Upgrade/Migration Guide:**  Several of my react-static projects are still on v5.9.12 because v6 was a little unstable out of the box for awhile and we were blocked by bugs we didn't have resources to fix ourselves. Some documentation already exists, but more assurances that all past use cases are definitely working for v7 as an up-front guarantee would go a long way.  Conceptually the upgrade path is pretty clear and not too much work, but there was some trepidation about upgrading react-static versions that's not totally gone yet. Recommendations on best practices for switching an existing project to react-static from another SSR framework would also drive adoption.  

I've found the above to be stumbling blocks at times when trying to increase usage of react-static within an organization. Once these were overcome though, feedback has been universally positive from both react devs with limited SSR experience and from backend devs with no react experience, which feels like react-static's sweet spot.

I think what exists for new users now is already good, but making it even better with a little more polish would be a quantitative addition that makes a qualitative difference.

I'm very excited to see what's next for react-static!

**Comment by SleeplessByte** [comment]
Hello everyone! 

I am working on releasing the maintenance release 7.2.0, which will take some effort from my side as I need to properly set up all the workspaces on the build machine. Worry not, this is coming!

**Comment by Vinnl** [comment]
Hmm, just found this issue, and especially:

>  consider partially upgrading some to TypeScript
> ...
> have proper tests for the plugins (!!)

What might be useful to know: the TypeScript plugin was transferred to this repo, but originally lived at https://gitlab.com/Vinnl/react-static-plugin-typescript/

Most importantly: it itself was written in TypeScript, and had unit tests. Thus, it might be helpful in those areas :)

As for your initial questions:

- :bug: #1203 is still stumping me, but it's probably not an easy fix for the same reason.
- :unicorn: Nothing really, I think.
- :gem: I'm really looking forward to the external type definitions :)

Thanks for taking up the baton!

**Comment by simonwalsh** [comment]
Amazing list @SleeplessByte! Thanks for doing this :)

**Comment by SleeplessByte** [comment]
React Static 8 is well underway. We've been dealing with other internal affairs unrelated to react static, but luckily there have been some amazing contributors. Thank you everyone for making this happen.

**Comment by nddery** [comment]
Thank you @SleeplessByte and everyone else involved! Was wondering if the progress toward React Static 8 was tracked somewhere - maybe in the [Triage project](https://github.com/react-static/react-static/projects/3) ? Happy to help tackle some of the remaining issues or features.

**Comment by SleeplessByte** [comment]
It's coming in the "react static 8" project, but I haven't had the time to document everything. I'll publish the first basic stuff to the `next` tag soon, and I'll tag you so you'll know :)
